### PR TITLE
MULE-16987: Interception API: Parameters of smart connector not intercepted

### PR DESCRIPTION
**Pull request by emariotti3**
…cepted.

### CODE REVIEW

**Comment by emariotti3** [comment]
--validate

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/BeanDefinitionFactory.java [hunk 1]
```diff
@@ -213,7 +213,7 @@ private BeanDefinition resolveComponent(ComponentModel parentComponentModel, Spr
             addAnnotation(ANNOTATION_NAME, componentModel.getIdentifier(), componentModel);
             // We need to use a mutable map since spring will resolve the properties placeholder present in the value if needed
             // and it will be done by mutating the same map.
-            addAnnotation(ANNOTATION_PARAMETERS, new HashMap<>(componentModel.getParameters()), componentModel);
+            addAnnotation(ANNOTATION_PARAMETERS, new HashMap<>(componentModel.getFilteredParameters()), componentModel);
```
Comment (elrodro83) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/BeanDefinitionFactory.java]: better to make a check by the identifier of the component and filtering the params here that spreading that logic to the ComponentModel.

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/model/extension/xml/MacroExpansionModuleModel.java [hunk 2]
```diff
@@ -297,6 +299,8 @@ private ComponentModel createModuleOperationChain(ComponentModel operationRefMod
     processorChainBuilder.addParameter("moduleOperation", operationModel.getName(), false);
     Map<String, String> propertiesMap = extractProperties(configRefName);
     Map<String, String> parametersMap = extractParameters(operationRefModel, operationModel.getAllParameterModels());
+    parametersMap.forEach((k, v) -> processorChainBuilder.addParameter(k, v, false));
```
Comment (elrodro83) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/model/extension/xml/MacroExpansionModuleModel.java]: is this still needed?

File: core/src/main/java/org/mule/runtime/core/internal/processor/interceptor/ReactiveInterceptorAdapter.java [hunk 3]
```diff
@@ -83,7 +85,7 @@ public ReactiveProcessor apply(Processor component, ReactiveProcessor next) {
     final ProcessorInterceptor interceptor = interceptorFactory.get();
     Map<String, String> dslParameters = (Map<String, String>) ((Component) component).getAnnotation(ANNOTATION_PARAMETERS);
 
-    ReactiveProcessor interceptedProcessor = doApply(component, next, componentLocation, interceptor, dslParameters);
+    ReactiveProcessor interceptedProcessor = doApply(component, next, componentLocation, interceptor, dslParameters.entrySet().stream().filter(param -> !(param.getKey().equals("moduleOperation") || param.getKey().equals("moduleName"))).collect(Collectors.toMap(p -> p.getKey(), p -> p.getValue())));
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/processor/interceptor/ReactiveInterceptorAdapter.java]: have a singleton list with these parameter names and check with `contains`

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/model/extension/xml/MacroExpansionModuleModel.java (hunk 2 above)
Comment (emariotti3) [reply modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/model/extension/xml/MacroExpansionModuleModel.java]: yes, this adds the parameters that were missing and that need to be intercepted

File: core/src/main/java/org/mule/runtime/core/internal/processor/interceptor/ReactiveInterceptorAdapter.java [hunk 4]
```diff
@@ -43,11 +45,10 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Optional;
+import java.util.*;
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/processor/interceptor/ReactiveInterceptorAdapter.java]: no * imports

File: core/src/main/java/org/mule/runtime/core/internal/processor/interceptor/ReactiveInterceptorAdapter.java [hunk 5]
```diff
@@ -83,7 +84,9 @@ public ReactiveProcessor apply(Processor component, ReactiveProcessor next) {
     final ProcessorInterceptor interceptor = interceptorFactory.get();
     Map<String, String> dslParameters = (Map<String, String>) ((Component) component).getAnnotation(ANNOTATION_PARAMETERS);
 
-    ReactiveProcessor interceptedProcessor = doApply(component, next, componentLocation, interceptor, dslParameters);
+    List macroExpansionInternalParams = unmodifiableList(Arrays.asList("moduleName", "moduleOperation"));
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/processor/interceptor/ReactiveInterceptorAdapter.java]: have this list as a static field

File: core/src/main/java/org/mule/runtime/core/internal/processor/interceptor/ReactiveInterceptorAdapter.java [hunk 6]
```diff
@@ -83,7 +84,9 @@ public ReactiveProcessor apply(Processor component, ReactiveProcessor next) {
...
+
+    ReactiveProcessor interceptedProcessor = doApply(component, next, componentLocation, interceptor, dslParameters.entrySet().stream().filter(param -> !macroExpansionInternalParams.contains(param.getKey())).collect(Collectors.toMap(p -> p.getKey(), p -> p.getValue())));
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/processor/interceptor/ReactiveInterceptorAdapter.java]: static import of `toMap`

**Pull request merged**